app/models/name.rb

```diff
@@ -408,33 +408,95 @@ class Name < AbstractModel
     end
   end
 
-  scope :with_rank,
-        ->(rank) { where(rank: Name.ranks[rank]) if rank }
+  include CreatedUpdatedScopes
 
+  scope :of_lichens, -> { where(Name[:lifeform].matches("%lichen%")) }
+  scope :not_lichens, -> { where(Name[:lifeform].does_not_match("% lichen %")) }
+  scope :deprecated, -> { where(deprecated: true) }
   scope :not_deprecated, -> { where(deprecated: false) }
-
+  scope :with_description, -> { where.not(description_id: nil) }
+  scope :without_description, -> { where(description_id: nil) }
   ### Module Name::Spelling
   scope :with_correct_spelling, -> { where(correct_spelling_id: nil) }
-
-  # For a glitch discovered in the wild:
+  scope :with_incorrect_spelling, -> { where.not(correct_spelling_id: nil) }
   scope :with_self_referential_misspelling, lambda {
     where(Name[:correct_spelling_id].eq(Name[:id]))
   }
-
+  scope :with_synonyms, -> { where.not(synonym_id: nil) }
+  scope :without_synonyms, -> { where(synonym_id: nil) }
+  ### Module Name::Create
+  # Get list of Names that are potential matches when creating a new name.
+  # Takes results of Name.parse_name.  Used by NameController#create_name.
+  # Three cases:
+  #
+  #   1. group with author       - only accept exact matches
+  #   2. nongroup with author    - match names with correct author or no author
+  #   3. any name without author - ignore authors completely when matching names
+  #
+  # If the user provides an author, but the only match has no author, then we
+  # just need to add an author to the existing Name.  If the user didn't give
+  # an author, but there are matches with an author, then it already exists
+  # and we should just ignore the request.
+  #
+  scope :matching_desired_new_name, lambda { |parsed_name|
```

Comment:
This is a fairly specialized query.  At very least it needs to state clearly in the name of the scope that it is working off of a "parsed_name" -- a very specialized class.  Secretly, I would like to remove all of that stuff -- the name parser, etc.  It's confusing as hell, and no one would ever use it to write new code.  I can't imagine GraphQL ever using it.  Hmmm.  Except maybe if you are trying to imitate perfectly the existing code... but that's one of the huge benefits to a rewrite-from-scratch... you don't need to imitate bad legacy code!!  I also don't like the idea of moving specialized complex code out of the module where it is used.  Unless this is actually used by multiple modules, but then the refactor is suspect for other reasons, but I suppose we just do the best we can with what we have... so fine, I guess it was a mess before, it is allowed to be a (different kind of) mess after.

Reply:
Good to know, about name parser. I'd be interested to hear if you've pictured a better way to do what it does, just to start thinking about it. It's a large thicket of code that i haven't even begun to try to find my way around. I just assumed it was key to the obscure workings of Name.

How about this... since scopes don't have to be at the top of the model file, I will just move these two back to where they were, in Create. I'd like to maybe keep them as scopes though — once you get used to them, `scope` makes the purpose of the method more obvious at a glance.

Reply:
Whoops, ok. Because `module Name::Create` is currently an _extendable_ module generating only class methods, rather than an _includable_ module, a `scope` cannot be declared there without refactoring the module. For now I'm moving the scope back to the Name class scopes, renaming it to make the parsed_name obvious, and of course it is plenty annotated.

Reply:
Just tried refactoring it as an _includable_ module, and moving it to `Name::Create` still did not work. I think we're stuck with all scopes in the model main file.

Reply:
No worries.  But do still consider renaming it with "parsed_name" in it somewhere -- "matching_parsed_name"?  That should help convince anyone reading through the scopes that unless they are using NameParse this is not the scope to use.  And it should emphasize how specialized this scope is.

Reply:
I had renamed it `matching_desired_new_parsed_name`. Not the snappiest!

And I'm definitely in favor of changing ParsedName as you said, "so that it only accepts existing names, and if the user wants to create a new name, make them create it separately". I kind of assumed that's what it did... I'd like to do that PR soon.